debezium-demo/demo.py

```diff
@@ -0,0 +1,163 @@
+import deephaven.ConsumeCdc as cc
+import deephaven.ConsumeKafka as ck
+import deephaven.ProduceKafka as pk
+import deephaven.Types as dh
+from deephaven import Aggregation as agg, as_list
+from deephaven import PythonFunction as pyfun
+import jpy
+
+def tmapfun(x):
+    return pyfun(x, 'io.deephaven.engine.table.Table')
+
+server_name = 'mysql'
+db_name='shop'
+
+kafka_base_properties = {
+    'bootstrap.servers' : 'redpanda:9092',
+    'schema.registry.url' : 'http://redpanda:8081',
+}
+
+
+def make_cdc_table(table_name:str):
+    return  cc.consumeToTable(
+        kafka_base_properties,
+        cc.cdc_short_spec(server_name,
+                          db_name,
+                          table_name)
+    )
+
+users = make_cdc_table('users')
+items = make_cdc_table('items')
+purchases = make_cdc_table('purchases')
+
+consume_properties = {
+    **kafka_base_properties,
+    **{
+        'deephaven.partition.column.name' : '',
+        'deephaven.timestamp.column.name' : '',
+        'deephaven.offset.column.name' : ''
+    }
+}
+
+pageviews = ck.consumeToTable(
+    consume_properties,
+    topic = 'pageviews',
+    key = ck.IGNORE,
+    value = ck.json([ ('user_id', dh.long_),
+                      ('url', dh.string),
+                      ('channel', dh.string),
+                      ('received_at', dh.datetime) ]),
+    table_type = 'append'
+)
+
+pageviews_stg = pageviews \
+    .updateView(
+        'url_path = url.split(`/`)',
+        'pageview_type = url_path[1]',
+        'target_id = Long.parseLong(url_path[2])'
+    ).dropColumns('url_path')
+
+purchases_by_item = purchases.aggBy(
+    as_list([
+        agg.AggSum('revenue = purchase_price'),
+        agg.AggCount('orders'),
+        agg.AggSum('items_sold = quantity')
+    ]),
+    'item_id'
+)
+
+pageviews_by_item = pageviews_stg \
+    .where('pageview_type = `products`') \
+    .countBy('pageviews', 'item_id = target_id')
+
+item_summary = items \
+    .view('item_id = id', 'name', 'category') \
+    .naturalJoin(purchases_by_item, 'item_id') \
+    .naturalJoin(pageviews_by_item, 'item_id') \
+    .dropColumns('item_id') \
+    .updateView('conversion_rate = orders / (double) pageviews')
+
+top_5_pageviews = item_summary \
+    .sortDescending('pageviews') \
+    .head(5)
+
+minute_in_nanos = 60 * 1000 * 1000 * 1000
+
+# Once deephaven-core PR #1843 is merged, we can remove the need for explicitly using L below
+# by defining a minute_in_nanos python variable, which would be interpolated with the right type,
+# and replacing `60L*second_in_nanos` by `minute_in_nanos`.
+profile_views_per_minute_last_10 = pageviews_stg \
+    .updateView(
+        'received_at_nanos = nanos(received_at)',
+        'received_at_minutes = received_at_nanos - received_at_nanos % minute_in_nanos'
+    ).where(
+        'pageview_type = `profiles`',
+        'nanos(DateTime.now()) < received_at_nanos + 10*minute_in_nanos'
```

Comment:
If you want to have things that actually update with time, you need to use window check; otherwise this is not a generally reliable way to use the engine, because now() can change out form under it.